Improvement suggestion for large circuitbreaker database

M1ch43lV opened the pull request:
Because of errors when doing the pull request to download the json data via web interface:
`{"code":8,"message":"grpc: received message larger than max (974495683 vs. 33554432)","details":[]}`
Or getting the error 429 for too many request, when using the command 
`wget -qO- http://localhost:9235/api/forwarding_history` 
I decided to extract the data directly from the circuitbreaker database using sqlite3.

The new file is called `channel_utilization_sqlite.sh`. It will make use of second file called `randomize-data-fast.sh` for faster randomization of channel_in and channel_out id.

M1ch43lV committed:
Create channel_utilization_sqlite.sh

In case of a very large database, so that the api routine breaks, process circuitbreaker.db locally with the help of sqlite.

M1ch43lV committed:
Faster randomization for larger databases

Faster randomization for larger databases (>1M entries)

M1ch43lV committed:
Update README.md

Addendum added for large circuitbreaker database, if the API request is failing

M1ch43lV committed:
Update README.md

Some typos corrected

carlaKC commented on channel-utilization/channel_utilization_sqlite.sh:
```diff
@@ -0,0 +1,52 @@
+process_batch() {
+    local offset=$1
+    sqlite3 -csv -noheader circuitbreaker.db <<EOF
+SELECT add_time, incoming_channel, outgoing_channel, incoming_amt_msat, outgoing_amt_msat,
```
Would you mind updating this to add `resolved_time` - just added this in the other script in #3 because I realized we're also going to need that value!

carlaKC commented on forwarding-history/randomize-data-fast.sh:
```diff
@@ -0,0 +1,88 @@
+  }
+  close(random_values_file)
+}
+function get_next_random() {
```
One of the goals for randomization is to have a consistent mapping for `channel_id` (in/out) -> `random_value` - we want to hide the `channel_id` because it contains information about the channel but we want to keep the traffic pattern the same so that we can look at the amount of htlcs that are being forwarded over a specific channel. 

If I'm reading this correctly, we're not doing that mapping like we do in the other randomization script so we'll lose the traffic patterns here?

carlaKC commented on forwarding-history/randomize-data-fast.sh:
```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
```
I think that this can be moved into the `channel-utilization` directory 👍

carlaKC reviewed (commented):
Thank you so much for the PR! 

Two asks for this one 🙏 
1. Could you add `resolved_time` as in #3 (oversight in the original script)
2. Randomization of channel ids needs to preserve mappings so we can preserve traffic patterns

Thank you!

M1ch43lV committed:
Update channel_utilization_sqlite.sh

1. Perform sqlite3 integrity check of circuitbreaker.db
2. Added resolved_time_ns

M1ch43lV committed:
Delete forwarding-history/randomize-data-fast.sh

Move file to directory channel-utilization

M1ch43lV committed:
Create randomize-data-fast.sh in directory channel-utilization

Added a lookup file for the pseudo-randomization of channel_ids

M1ch43lV committed:
Merge branch 'carlaKC:master' into master

M1ch43lV committed:
Update README.md

M1ch43lV commented:
1. `resolved_time` added
2. Integrity check of circuitbreaker.db included
3. Moved `randomized-data-fast.sh` in directory channel-utilization
4. Modified `randomized-data-fast.sh` for pseudo-randomization of channel_ids to preserve traffic patterns
5. Modified `randomized-data-fast.sh` to create a lookup-file for the pseudo-randomized channel_ids

carlaKC commented on channel-utilization/README.md:
````diff
@@ -1,26 +1,79 @@
+
+   Example of a copy command depending on where `circuitbreaker.db` and the `channel-utilization` directory are located:
+   ```sh
+   cp /home/circuitbreaker/.circuitbreaker/circuitbreaker.db /home/admin/ln-data/channel-utilization/.
````
nit: this is specific to where you cloned `ln-data`

let's update to just dropping it in the working directory? 
`cp /home/circuitbreaker/.circuitbreaker/circuitbreaker.db .`

Would also be great to make a note that the database may be in a difference place on different OS-es (eg mac is: `~/Library/Application\ Support/Circuitbreaker/circuitbreaker.db`)

carlaKC reviewed (approved):
Great stuff - just one nitpick about database location.
